cpp/zeder_to_zotero_importer.cc

```diff
@@ -0,0 +1,432 @@
+/** \brief Utility to automatically update the Zotero Harvester configuration from Zeder.
+ *  \author Madeeswaran Kannan
+ *
+ *  \copyright 2020 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <iostream>
+#include <unordered_set>
+#include <cstdio>
+#include <cstdlib>
+#include "FileUtil.h"
+#include "IniFile.h"
+#include "util.h"
+#include "ZoteroHarvesterConfig.h"
+#include "ZoteroHarvesterUtil.h"
+#include "ZoteroHarvesterZederInterop.h"
+
+
+namespace {
+
+
+using namespace ZoteroHarvester;
+
+
+[[noreturn]] void Usage() {
+    std::cerr << "Usage: " << ::progname << " [options] config_file_path mode zeder_flavour zeder_ids fields_to_update\n"
+              << "\n"
+              << "\tOptions:\n"
+              << "\t[--min-log-level=log_level]     Possible log levels are ERROR, WARNING (default), INFO and DEBUG\n"
+              << "\t[--overwrite-on-import]         Overwrite existing entries/sections when performing an import\n"
+              << "\n"
+              << "\tconfig_file_path                Path to the Zotero Harvester config file\n"
+              << "\tmode                            Either IMPORT or UPDATE\n"
+              << "\tzeder_flavour                   Either IXTHEO or KRIMDOK\n"
+              << "\tzeder_ids                       Comma-separated list of Zeder entry IDs to import/update.\n"
+              << "\t                                Special-case for updating: Use '*' to update all entries found in the config that belong to the Zeder flavour\n"
+              << "\tfields_to_update                Comma-separated list of the following fields to update: \n"
+              << "\t                                \tONLINE_PPN, PRINT_PPN, ONLINE_ISSN, PRINT_ISSN, EXPECTED_LANGUAGES, ENTRY_POINT_URL, UPLOAD_OPERATION, UPDATE_WINDOW.\n"
+              << "\t                                Ignored when importing entries (all importable fields will be imported).\n\n";
+    std::exit(EXIT_FAILURE);
+}
+
+
+struct CommandLineArgs {
+    enum Mode { INVALID, IMPORT, UPDATE };
+
```

Comment:
We use public to separate the type declarations from the field declarations.  Alternatively you can declare the enum field on the same line as the Mode type.